kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -1667,6 +1672,16 @@ internal class CodeGeneratorVisitor(
         }
     }
 
+    private fun isCompanionParentClass(value: IrTypeOperatorCall, dstClass: IrClass): Boolean {
+        var result = true
+        if (dstClass.isCompanion) {
+            (dstClass.parent as? IrClass)?.symbol?.let {
+                result = value.argument.type == it
+            }
+        }
+        return result
+    }
```

Comment:
I don't really understand what was the intention, but the result doesn't seem correct:
```kotlin
class Foo {
    companion object
}

fun main() {
    val foo: Any = Foo
    println(foo is Foo.Companion)
}
```
prints `false` now, which is wrong.

Reply:
Wrong `if` judgment, only covering specific `NSObject` test cases.